ng_package/angular_package_format.bzl

```diff
@@ -2,56 +2,427 @@ load("//ng_project/config:compilation_mode.bzl", "partial_compilation_transition
 load("@aspect_rules_js//js:libs.bzl", "js_lib_helpers")
 load("@aspect_rules_js//js:providers.bzl", "JsInfo")
 
+# Prints a debug message if "--define=VERBOSE_LOGS=true" is specified.
+def _debug(vars, *args):
+    if "VERBOSE_LOGS" in vars.keys():
+        print("[ng_package.bzl]", args)
+
+_DEFAULT_NG_PACKAGER = "//ng_package/packager:bin"
+_DEFAULT_ROLLUP_CONFIG_TMPL = "//ng_package/rollup:rollup.config.js"
+_DEFAULT_ROLLUP = "//ng_package/rollup:bin"
+
+_NG_PACKAGE_MODULE_MAPPINGS_ATTR = "ng_package_module_mappings"
+
+WELL_KNOWN_EXTERNALS = [
+    "@angular/animations",
+    "@angular/animations/browser",
+    "@angular/animations/browser/testing",
+    "@angular/common",
+    "@angular/common/http",
+    "@angular/common/http/testing",
+    "@angular/common/testing",
+    "@angular/common/upgrade",
+    "@angular/compiler",
+    "@angular/core",
+    "@angular/core/testing",
+    "@angular/elements",
+    "@angular/forms",
+    "@angular/localize",
+    "@angular/localize/init",
+    "@angular/platform-browser",
+    "@angular/platform-browser/animations",
+    "@angular/platform-browser/testing",
+    "@angular/platform-browser-dynamic",
+    "@angular/platform-browser-dynamic/testing",
+    "@angular/platform-server",
+    "@angular/platform-server/init",
+    "@angular/platform-server/testing",
+    "@angular/router",
+    "@angular/router/testing",
+    "@angular/router/upgrade",
+    "@angular/service-worker",
+    "@angular/service-worker/config",
+    "@angular/upgrade",
+    "@angular/upgrade/static",
+    "rxjs",
+    "rxjs/operators",
+    "tslib",
+]
+
+def _write_rollup_config(
+        ctx,
+        root_dir,
+        metadata_arg,
+        side_effect_entry_points,
+        dts_mode):
+    filename = "_%s_%s.rollup.conf.js" % (ctx.label.name, "dts" if dts_mode else "fesm")
+    config = ctx.actions.declare_file(filename)
+
+    mappings = dict()
+    all_deps = ctx.attr.srcs
+    for dep in all_deps:
+        if hasattr(dep, _NG_PACKAGE_MODULE_MAPPINGS_ATTR):
+            for k, v in getattr(dep, _NG_PACKAGE_MODULE_MAPPINGS_ATTR).items():
+                if k in mappings and mappings[k] != v:
+                    fail(("duplicate module mapping at %s: %s maps to both %s and %s" %
+                          (dep.label, k, mappings[k], v)), "deps")
+                mappings[k] = v
+
+    externals = WELL_KNOWN_EXTERNALS + ctx.attr.externals
+
+    # Pass external & globals through a templated config file because on Windows there is
+    # an argument limit and we there might be a lot of globals which need to be passed to
+    # rollup.
+
+    ctx.actions.expand_template(
+        output = config,
+        template = ctx.file.rollup_config_tmpl,
+        substitutions = {
+            "TMPL_banner_file": "\"%s\"" % ctx.file.license_banner.path if ctx.file.license_banner else "undefined",
+            "TMPL_module_mappings": str(mappings),
+            # TODO: Determine node_modules_root
```

Comment:
Probably can remove this in a follow-up; I don't think we need the whole node module plugin.

Reply:
Same for the mappings above; but let's keep for now and remove when we are sure